### PR TITLE
Allow NONE for item 7 in GCONPROD

### PR DESCRIPTION
**Pull request by plgbrts**
This PR is to allow for default value `NONE` of item 7 of `GCONPROD`. So far only `RATE` was allowed for item 7 (see https://github.com/OPM/opm-simulators/pull/4593)

### CODE REVIEW

**Comment by plgbrts** [comment]
jenkins build this opm-simulators=4658 please

**Comment by plgbrts** [comment]
Below  we compare OPM-Flow with a commercial reservoir simulator for both options (`NONE` and` RATE`) for  item 7. 
The simulations are in agreement.

In file `0A1_GRCTRL_LRAT_ORAT_BASE_MODEL2_STW_7RATE_OPM.DATA` :
```
GCONPROD 
-- for group PROD: target is LRATE=2500; constraints are ORATE, WRATE and GRATE, 
-- if item 7 is NONE then constraints are ignored 
-- if item 7 is RATE the constraints needs to be honored
--   1     2      ORATE  WRATE GRATE LRATE    7
  'PROD' 'LRAT'   1000. 1*     1.E6  2500.  'RATE' 'YES' 1*  /  --constraint ORAT 1000 honoured 
  'WGRP1' 'FLD'   900.  1*     1.E6  2400.  'RATE' 'YES' 1*  / 
  'WGRP2' 'FLD'   900.  1*     1.E6  2400.  'RATE' 'YES' 1*  / 
/
```
In `0A1_GRCTRL_LRAT_ORAT_BASE_MODEL2_STW_7NONE_OPM.DATA`:
```
GCONPROD 
--   1     2      ORATE  WRATE GRATE LRATE    7
  'PROD' 'LRAT'   1000. 1*     1.E6  2500.  'NONE' 'YES' 1* /  --constraint ORAT 1000 ignored
  'WGRP1' 'FLD'   900.  1*     1.E6  2400.  'RATE' 'YES' 1*  / 
  'WGRP2' 'FLD'   900.  1*     1.E6  2400.  'RATE' 'YES' 1*  / 
```
 The `*_ECL.DATA` are identical to `*_OPM.DATA` and run with the commercial simulator

image

File: src/opm/input/eclipse/Schedule/KeywordHandlers.cpp [hunk 1]
```diff
@@ -564,27 +564,69 @@ File {} line {}.)", wname, location.keyword, location.filename, location.lineno)
...
+                                 production.production_controls += static_cast<int>(Group::ProductionCMode::WRAT);
+                            if (!apply_default_gas_target)
+                                 production.production_controls += static_cast<int>(Group::ProductionCMode::GRAT);
+                            if (!apply_default_liquid_target)
+                                 production.production_controls += static_cast<int>(Group::ProductionCMode::LRAT);
+                        }
+                    }
 
                     if (!apply_default_resv_target)
                         production.production_controls += static_cast<int>(Group::ProductionCMode::RESV);
```
Comment (totto82) [inline src/opm/input/eclipse/Schedule/KeywordHandlers.cpp:632]: Why is the same logic not applied to RESV?

**Review by totto82 — COMMENTED** [review]
Thanks. My only concern is related to RESV. Should RESV follow the same logic as the rest of the targets?

File: src/opm/input/eclipse/Schedule/KeywordHandlers.cpp (hunk 1 above)
Comment (plgbrts) [reply src/opm/input/eclipse/Schedule/KeywordHandlers.cpp:632]: According to the commercial simulator description the procedure of exceeding violation of the RESV limit (specified in item 14) is always RATE and thus any setting of item 7 should be ignored to be consistent with the commercial simulator.

**Comment by totto82** [comment]
Thanks!

**Pull request merged**